runtime/src/constants.rs

```diff
@@ -35,6 +38,40 @@ pub const DAYS: BlockNumber = HOURS * 24;
 // 1 in 4 blocks (on average, not counting collisions) will be primary babe blocks.
 pub const PRIMARY_PROBABILITY: (u64, u64) = (1, 4);
 
+parameter_types! {
+    pub const ProposalsLockId: LockIdentifier = [5; 8];
+    pub const StorageWorkingGroupLockId: LockIdentifier = [6; 8];
+    pub const ContentWorkingGroupLockId: LockIdentifier = [7; 8];
+    pub const ForumGroupLockId: LockIdentifier = [8; 8];
+}
+
+lazy_static! {
+    // pairs of allowed lock combinations
+    pub static ref ALLOWED_LOCK_COMBINATIONS: BTreeSet<(LockIdentifier, LockIdentifier)> = [
```

Comment:
This is a good idea to maintain such a table! However, I think that we could get away with much less coding. I suggest maintaining pairs `(LockId, Vec<LockId>)`. This would quickly show what locks allow concurrency. And also, it is easy to set up - because it would be just lines from the 'staking reusability relationships' table of the handbook. What do you think?

Reply:
That was also my initial idea, but I didn't like the necessity to iterate through vector on each check. Binary tree set lookup should be more performance effective. On the other hand, it is definitely easier to define/inspect the table with a vector, as you suggested. 

I will rewrite it so we can definite the table like `(LockId, Vec<LockId>)`, but it's immediately recalculated to `BTreeSet` for faster lookup.